plugins/commands/ssh_config/command.rb

```diff
@@ -32,6 +32,11 @@ def execute
           ssh_info = machine.ssh_info
           raise Vagrant::Errors::SSHNotReady if ssh_info.nil?
 
+          if not options[:host].nil? and
+             not options[:host].to_s.eql?(machine.name.to_s)
+            next
+          end
+
```

Comment:
Hi,
I've try to simplify it into `next if options[:host] != machine.name`, but failed,
see [Travis-CI Here](https://travis-ci.org/mitchellh/vagrant/builds/165839731#L335-L339)

the reason I put `not options[:host].nil?` in condition block is trying to avoid returning empty string if `--host XXX` not match

thanks for your quick review :+1: